tdns/dnshandler.go

```
@@ -86,6 +86,17 @@ func createHandler(conf *Config) func(w dns.ResponseWriter, r *dns.Msg) {
 			log.Printf("Zone %s %s request from %s", qname, dns.TypeToString[qtype], w.RemoteAddr())
 
 			if zd, ok := Zones.Get(qname); ok {
+				if zd.Error {
+					if zd.ErrorType != RefreshError || zd.RefreshCount == 0 {
+						log.Printf("DnsHandler: Qname is %q, which is a known zone, but it is in %s error state: %s",
+							qname, zd.ErrorType, zd.ErrorMsg)
+						m := new(dns.Msg)
+						m.SetRcode(r, dns.RcodeServerFailure)
+						w.WriteMsg(m)
```

golangci-lint (1.62.2)

Error return value of `w.WriteMsg` is not checked

(errcheck)